rholang/src/main/scala/coop/rchain/rholang/interpreter/Reduce.scala

```diff
@@ -192,23 +217,17 @@ object Reduce {
           case _ => s.unit
         }
 
-      def mkJob[A](
-          terms: Seq[A],
-          handler: (A, Blake2b512Random) => M[Unit],
-          startIdx: Int
-      ): M[List[Unit]] =
-        Parallel.parTraverse(terms.zipWithIndex.toList) {
-          case (term, idx) => handler(term, split(rand, startIdx, idx))
-        }
-
       List(
-        mkJob(par.sends, mkTermHandler[Send](evalExplicit), 0),
-        mkJob(par.receives, mkTermHandler[Receive](evalExplicit), 1),
-        mkJob(par.news, mkTermHandler[New](evalExplicit), 2),
-        mkJob(par.matches, mkTermHandler[Match](evalExplicit), 3),
-        mkJob(par.bundles, mkTermHandler[Bundle](evalExplicit), 4),
-        mkJob(filteredExprs, exprHandler, 5)
-      ).parSequence.as(Unit)
+        EvalJob(par.sends, mkTermHandler[Send](evalExplicit)),
+        EvalJob(par.receives, mkTermHandler[Receive](evalExplicit)),
+        EvalJob(par.news, mkTermHandler[New](evalExplicit)),
+        EvalJob(par.matches, mkTermHandler[Match](evalExplicit)),
+        EvalJob(par.bundles, mkTermHandler[Bundle](evalExplicit)),
+        EvalJob(filteredExprs, exprHandler)
+      ).zipWithIndex
```

Comment:
Honeslty, I don't see any benefit from introducing the EvalJob. If you wanted to get rid of the hardcoded ints, I think you could've done:

```scala
        mkJob(par.sends, mkTermHandler[Send](evalExplicit), _),
        mkJob(par.receives, mkTermHandler[Receive](evalExplicit), _),
        mkJob(par.news, mkTermHandler[New](evalExplicit), _),
        mkJob(par.matches, mkTermHandler[Match](evalExplicit), _),
        mkJob(par.bundles, mkTermHandler[Bundle](evalExplicit), _),
        mkJob(filteredExprs, exprHandler, _)
      )
.zipWithIndex
.map(i, f -> f(i))
.parSequence.as(Unit)

```